### PR TITLE
make schema

### PR DESCRIPTION
**Pull request by mgmediaweb**
# Features

- Makes the file schema_based_on_diagram.sql with the queries to make tables.
- Makes tables for patients and treatments first, because they do not have any relations with other tables.
- Makes tables for medical_history, invoices, and invoice_items makes specific relations with tables.
- Follow the specification from the schema provided.

image

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    id SERIAL NOT NULL,
+    type VARCHAR(20),
+    name VARCHAR(100) NOT NULL,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_history (
+    id SERIAL NOT NULL,
+    admitted_at TIMESTAMP,
+    patient_id INT REFERENCES patients(id),
```
Comment (zieeco) [inline schema_based_on_diagram.sql]: - Please, kindly ensure your table `medical_histroy` is pluralized to `medical_histories`as this is a table that will contain many details about the treatments of the patients, so it's meant to be plural.

- Please, kindly add also the foreign key that references the treatment id from the treatments table.
e.g 

`ALTER TABLE medical_histories ADD FOREIGN KEY (treatment_id) REFERENCES treatments(id);`

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    status VARCHAR(20),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE invoices  (
+    id SERIAL NOT NULL,
+    total_amount DECIMAL(6,3),
+    generate_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT REFERENCES medical_history(id),
```
Comment (zieeco) [inline schema_based_on_diagram.sql]: - Please, kindly ensure your table `medical_histroy` is pluralized to `medical_histories`as this is a table that will contain many details about the treatments of the patients, so it's meant to be plural.
- Update the references `medical_history_id INT REFERENCES medical_histories(id)`

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    total_amount DECIMAL(6,3),
+    generate_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT REFERENCES medical_history(id), 
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE invoice_items (
+    id SERIAL NOT NULL,
+    unique_price DECIMAL(6,2),
```
Comment (zieeco) [inline schema_based_on_diagram.sql]: - Please, kindly ensure you've used the correct column name as `unit_price` and not `unique_price` as you've used.

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (zieeco) [inline schema_based_on_diagram.sql]: - Please, kindly create a join table for the `many-to-many` relationship between the `medical_histories` and `treatments` table and add the foreign key to reference the tables.
e.g

```sql
 CREATE TABLE medical_histories_has_treatments (
    medical_history_id int references medical_histories(id),
    treatment_id int reference treatments(id),
    );
```

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @mgmediaweb ,


Great job so far! 👏 
You've done welll, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice schema file. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (mgmediaweb) [reply schema_based_on_diagram.sql]: ok, I will make the change

File: schema_based_on_diagram.sql (hunk 3 above)
Comment (mgmediaweb) [reply schema_based_on_diagram.sql]: you are right!

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (mgmediaweb) [reply schema_based_on_diagram.sql]: Done!

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @mgmediaweb, 




Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**